shell/plugins/steve/worker/web-worker.advanced.js

```diff
@@ -241,33 +267,33 @@ const resourceWatcherActions = {
   'resource.change': (msg) => {
     const { type, id, data } = makeResourceProps(msg);
 
-    if (caches[type]) {
-      caches[type].change(data, () => workerActions.updateBatch(type, id, data));
-    } else {
-      workerActions.updateBatch(type, id, data);
+    if (!caches[type]) {
+      caches[type] = resourceCache(type);
     }
+
+    caches[type].change(data, () => workerActions.updateBatch(type, id, data));
   },
   // ToDo: SM create is functionally identical to change in the cache but the worker isn't supposed to know that hence the near-duplicate code
```

Comment:
these are identical now but they won't be once I'm making API requests in the worker so it makes sense to keep them separate since they're gonna diverge in the next PR